src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -4578,6 +4579,8 @@ public void Compact(Action<IOperationProgress> onProgress, CompactionResult resu
                 PathSetting compactPath = null;
                 PathSetting tempPath = null;
 
+                DocumentDatabase.ForTestingPurposes?.IndexCompaction?.Invoke();
```

Comment:
Use the `ForTestingPurposes` of the `IndexStore`

Reply:
And shouldn't it be after `RestartEnvironment`?

Reply:
Done